### ISSUE TITLE
Performance issue in data_provider/lanenet_data_feed_pipline.py

### ISSUE THREAD

**Issue by DLPerf** [MaybeShewill-CV/lanenet-lane-detection#520]
Hello! I've found a performance issue in data_provider/lanenet_data_feed_pipline.py: `dataset = dataset.batch(batch_size=batch_size, drop_remainder=True)`[(here)](https://github.com/MaybeShewill-CV/lanenet-lane-detection/blob/94be8d9a66c7bbda5ad99b10492bc406c8aee45f/data_provider/lanenet_data_feed_pipline.py#L307) should be called before `dataset.map(map_func=tf_io_pipline_tools.decode,num_parallel_calls=CFG.DATASET.CPU_MULTI_PROCESS_NUMS)`[(here)](https://github.com/MaybeShewill-CV/lanenet-lane-detection/blob/94be8d9a66c7bbda5ad99b10492bc406c8aee45f/data_provider/lanenet_data_feed_pipline.py#L284), `dataset.map(map_func=tf_io_pipline_tools.augment_for_train,num_parallel_calls=CFG.DATASET.CPU_MULTI_PROCESS_NUMS)`[(here)](https://github.com/MaybeShewill-CV/lanenet-lane-detection/blob/94be8d9a66c7bbda5ad99b10492bc406c8aee45f/data_provider/lanenet_data_feed_pipline.py#L289), `dataset.map(map_func=tf_io_pipline_tools.augment_for_test,num_parallel_calls=CFG.DATASET.CPU_MULTI_PROCESS_NUMS)`[(here)](https://github.com/MaybeShewill-CV/lanenet-lane-detection/blob/94be8d9a66c7bbda5ad99b10492bc406c8aee45f/data_provider/lanenet_data_feed_pipline.py#L294) and `dataset.map(map_func=tf_io_pipline_tools.normalize,num_parallel_calls=CFG.DATASET.CPU_MULTI_PROCESS_NUMS)`[(here)](https://github.com/MaybeShewill-CV/lanenet-lane-detection/blob/94be8d9a66c7bbda5ad99b10492bc406c8aee45f/data_provider/lanenet_data_feed_pipline.py#L298), which could make your program more efficient.

Here is [the tensorflow document](https://tensorflow.google.cn/guide/data_performance?hl=zh_cn#vectorized_mapping) to support it.

Besides, you need to check the function called in `map()`(e.g., `tf_io_pipline_tools.normalize` called in `map()`) whether to be affected or not to make the changed code work properly. For example, if `tf_io_pipline_tools.normalize` needs data with shape (x, y, z) as its input before fix, it would require data with shape (batch_size, x, y, z) after fix.

Looking forward to your reply. Btw, I am very glad to create a PR to fix it if you are too busy.

**Comment by MaybeShewill-CV** [comment]
@DLPerf You're welcome to make a PR. I will check it when got enough spare time. Work's busy. Sorry:)

**Comment by DLPerf** [comment]
OK, I'll try my best to make a PR. @MaybeShewill-CV

**Comment by MaybeShewill-CV** [comment]
@DLPerf 1.No obvious performance boost was found in my local test. Have you got any good test result yourself:) 2. normalize func works properly.

**Comment by MaybeShewill-CV** [comment]
@DLPerf  Have you got any good test result yourself:)

**Comment by codexsys-7** [comment]
Hey @DLPerf Please help me with my error, I was not able to train the model with my own parameters, it's prompting me an error to generate tf_records before training your model, and I'm not able to generate tf_records, would you please give an explanation or how did you generate tf_records to train the model.
Please for the love of God, Thanks in Advance...
